DNAnexus_to_igv/views.py

```diff
@@ -86,142 +122,201 @@ def nexus_search(request):
 
             search_form = Forms.SearchForm(request.POST)
 
             if search_form.is_valid():
                 clean_data = search_form.cleaned_data
-            
+
             sample_id = clean_data["sampleID"]
-            sample_id = str(sample_id).strip() # in case spaces
-            sample_id = sample_id.upper() # in case C/G/X is lower case
+            sample_id = str(sample_id).strip()  # in case spaces
 
             try:
-              # load in json with all bams and dx attributes needed to
-              # search and generate dx download links
-              # if json is not present in same dir it will raise IOError
-                json_file = os.path.join(os.path.dirname(__file__),
-                                                    "dx_002_bams.json")
+                # load in json with all bams and dx attributes needed to
+                # search and generate dx download links
+                # if json is not present it will raise IOError
+                json_file = os.path.join(
+                    os.path.dirname(__file__), "dx_002_bams.json")
+
                 with open(json_file) as json_file:
                     json_bams = json.load(json_file)
-            
-            except IOError:
-                messages.add_message(request,
-                                messages.ERROR,
-                                """An error has occured connecting with\
-                                DNAnexus to find samples, please contact\
-                                the bioinformatics team""",\
-                                extra_tags="alert-danger"
-                            )
-
-                return render(request, 'DNAnexus_to_igv/nexus_search.html', 
-                                context_dict)
-
-            # select bams matching sample id            
-            sample_bams = [v for k,v in json_bams.iteritems() if k.startswith(sample_id)]
+
+            except IOError as IOe:
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    """An error has occured connecting with\
+                    DNAnexus to find samples, please contact\
+                    the bioinformatics team""",
+                    extra_tags="alert-danger"
+                )
+                error_log.error(re.sub(
+                    r'\s+', ' ', """Failed to load sample list from JSON, most
+                    likely the JSON has not been generated with
+                    find_dx_002_bams.py"""
+                ))
+                logging.error(IOe)
+
+                return render(
+                    request, 'DNAnexus_to_igv/nexus_search.html', context_dict
+                )
+
+            # select bams matching sample id, return original entry from
+            # JSON by mtahcing against upper name and search term
+            sample_bams = [
+                v for k, v in json_bams.items() if sample_id.upper() in v[0]['bam_name'].upper()
```

Comment:
maybe use full name variable here to help comprehension
also i'd write:
```
[
    thing you extract
    for thing in things
    if condition
]
```

Reply:
also sensible, changed and added structure of output json to find bams script